### PR TITLE
Video transformers - added fixes to enable model load after saving and added HF model & space 

### PR DESCRIPTION
**Pull request by shivalikasingh95**
Hi, I have updated the video transformers example with links to model & space on Hugging Face hub. 

While trying to save and re-load the model, I faced few errors. Have added the fixes for those as well.

### CODE REVIEW

File: examples/vision/video_transformers.py [hunk 1]
```diff
@@ -279,6 +293,17 @@ def call(self, inputs, mask=None):
         proj_output = self.dense_proj(proj_input)
         return self.layernorm_2(proj_input + proj_output)
 
+    def get_config(self):
```
Comment (shivalikasingh95) [inline examples/vision/video_transformers.py:296]: Added to fix loading of model after saving.

File: examples/vision/video_transformers.py [hunk 2]
```diff
@@ -246,12 +249,23 @@ def compute_mask(self, inputs, mask=None):
...
+        )
+        return config
+
 
 """
 Now, we can create a subclassed layer for the Transformer.
 """
 
 
+@tf.keras.utils.register_keras_serializable(package="Custom", name="TransformerEncoder")
```
Comment (shivalikasingh95) [inline examples/vision/video_transformers.py:268]: Added this to avoid passing custom_objects dict to tf.keras.models.load_model()

**Comment by shivalikasingh95** [comment]
There's another issue I have noticed with this example.

**Problem description:**
After saving and re-loading the model, there's a big drop in its accuracy. However, I could see the drop in accuracy only if I saved it using TF SavedModel format. I didn't observe the accuracy drop issue if I save it using HDF5 format. 
Added screenshot for reference:
Screenshot from 2022-07-27 03-20-04

**Tried solutions**:
I tried a bunch of solutions listed in [this thread][2] but none of the below approaches helped:

1. loading model by passing 'custom_objects' dict parameter to tf.keras.models.load_model()
2. Registering custom layers of the model by using @tf.keras.utils.register_keras_serializable decorator to avoid passing 'custom_objects' dict parameter to tf.keras.models.load_model() at load time
3. Passing 'save_traces=True' to tf.keras.models.save_model() at the time of saving the model using TF savedmodel format
4. Specifying metrics as 'sparse_categorical_accuracy' instead of just 'accuracy' at the time of training the model as mentioned in [this issue][3] and then trying to save & load 
5. Loading model with compile=False as mentioned [here][4]
6. Tried using different versions of tensorflow - 2.8.2, 2.9.1 and even latest version available using tf-nightly (getting same issue with all tried versions)
7. Setting manual_variable_initialization(True) as mentioned [here][5] 

**What worked:**
However, if I define the model by Subclassing instead of using the Functional API, then the drop in accuracy doesn't occur on re-loading model after saving using TF SavedModel format.

**Question:**
Ideally, I don't want to change the Functional API based implementation of the model to a Subclassing based one. Hence, currently I have not added this as a fix as part of this PR.
However, I was not able to find any other alternative solution to this problem as well. Any suggestions on how can I solve this issue while preserving the original Functional API based implementation of the model ?

**Sample code**:
Also sharing full code to reproduce the issue in this [colab notebook][6]


  [2]: https://github.com/keras-team/keras/issues/4875
  [3]: https://stackoverflow.com/questions/64541824/keras-giving-low-accuracy-after-loading-model
  [4]: https://github.com/keras-team/keras/issues/4875#issuecomment-903000867
  [5]: https://github.com/keras-team/keras/issues/4875#issuecomment-296696536
  [6]: https://colab.research.google.com/drive/1NIu3rz2ulwKRuWPBz9SWUEz_mAehdPkp?usp=sharing

**Comment by shivalikasingh95** [comment]
Hi @fchollet, would be great if you could take a quick look at this PR whenever you find the time. It has minimal suggested changes.

**Comment by fchollet** [comment]
> Hi, I have updated the video transformers example with links to model & space on Hugging Face hub.

Thanks!

Please remove serialization support, it is not needed for this example and it adds complexity to the code example. If you want to reload a saved model, please use `model.save_weights()`/`load_weights()`. You do not need to store the model's architecture in serialized format since you have access to the model's code.

**Comment by shivalikasingh95** [comment]
> > Hi, I have updated the video transformers example with links to model & space on Hugging Face hub.
> 
> Thanks!
> 
> Please remove serialization support, it is not needed for this example and it adds complexity to the code example. If you want to reload a saved model, please use `model.save_weights()`/`load_weights()`. You do not need to store the model's architecture in serialized format since you have access to the model's code.

Hi @fchollet , makes sense. I had initially added that because I was trying to load the model for inference in a separate script where I didn't have access to the model's code.
But within the context of this example, it makes sense to keep the code simple as we already have access to the model's code.

Have removed the serialization support now. Let me know if you'd like me to fix anything else.

**Comment by pcoet** [comment]
@fchollet Can you have a look at this?

**Comment by divyashreepathihalli** [comment]
@shivalikasingh95 sorry this Pr has slipped through the cracks. Since Keras has changed from tf backend to multi-backend. the PR changes  would no longer be applicable. I will close this out. please feel free to open a new one. Thank you!
